packages/formstr-sdk/docs/design.md

```diff
@@ -0,0 +1,241 @@
+Formstr sdk is the tooling designed on top of the nostr protocol to help create forms that can be embedded into any website, software or usecase. This document outlines the basic api and functionality of the sdk.
+
+## API
+
+### Create a form
+
+a form can be created using the `createForm` function like:
+
+```js
+let formCredentials = formstrSDK.createForm(formSpec);
+```
+
+formspec: A form object that holds the information, example spec:
+
+```json
+{
+  "schemaVersion": "<link to schema>",
+  "name": "Pets Survey",
+  "description": "A survey about your pets",
+  "settings": {
+    "selfSignForms": false
+  },
+  "fields": [
+    {
+      "question": "What is the name of your pet?",
+      "answerType": "string"
+    },
+    {
+      "question": "wHAT DO YOU LOVE ABOUT YOUR PET?",
+      "answerType": "text"
+    },
+    {
+      "question": "What animal is your pet? ",
+      "answerType": "singleChoice",
+      "choices": [
+        {
+          "message": "cat"
+        },
+        {
+          "message": "dog"
+        },
+        {
+          "message": "other",
+          "otherMessage": true
+        }
+      ]
+    },
+    {
+      "question": "Now we need to know about your history with pets.\nallergies, what kinds of pet you had as a child",
+      "answerType": "label"
+    },
+    {
+      "question": "How long have you owned a pet for?",
+      "answerType": "number",
+      "numberConstraints": {
+        "max": 50,
+        "min": 0
+      }
+    }
+  ]
+}
+```
+
+### View Past Forms
+
+```js
+let pastForms = formstrSDK.getPastForms(userPrivateKey);
+```
+
+- userPrivateKey(optional): is a nostr private key, you can use an existing nostr private key, or generate a new one using the `nostr-tools` `generatePrivateKey` function.
+
+If the userPrivateKey is not provided the sdk will attempt to use the `window.nostr` functionality as outlined in [NIP-07](https://github.com/nostr-protocol/nips/blob/master/07.md)
+
+```json
+[
+  {
+    "formLink": "https://formstr.app/#/forms/:form_id",
+    "formId": "sfwsafsaf",
+    "formSecret": "dsfsdfsafsdf..."
+  },
+  ...
+]
+```
+
+### Save form to nostr
+
+All forms are backed up to nostr automatically with the private key/nip-07 used to initialize the client, but if a user wants to save a form manually, they can do:
+
+```js
+formstrSDK.saveForm(formPrivateKey);
```

Comment:
`formstrSDK.saveFormLocally();`

Reply:
added in the description, not sure if a saveFormLocally is required as a seperate api